iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/GpuIcebergPartitioner.scala

```diff
@@ -69,95 +99,43 @@ class GpuIcebergPartitioner(val spec: PartitionSpec,
     if (input.numRows() == 0) {
       return Seq.empty
     }
-
-    val numRows = input.numRows()
+    val inputColumnIndices = makeInputIndices(input.numCols())
```

Comment:
@liurenjie1024  unlike `keyColIndices`, we compute `inputColumnIndices` for each partition call,  could we get input columns number via:
`inputSchema.columns.size` or `dataSparkType.fields.size`.
Is the following right?
input.numCols() == inputSchema.columns.size == dataSparkType.fields.size

Reply:
Yes, it's correct.